### ISSUE TITLE
🚀Feature request: Param making the artifact a temporary one (i.e. available only during the current workflow)

### ISSUE THREAD

**Issue by valado** [actions/upload-artifact#121]
Issue type: Enhancement

**### Is your proposal related to a problem?**

So far the only way I found to share files/folders between jobs within a workflow was artifacts.
Problem is someone has to clean them afterwards or get their whole dev process blocked by reaching the storage limits.

**### Describe the solution you'd like**

Possible solutions
- add a param for upload-artifact (e.g. publish, defaults true) If publish is false, the artifact doesn't get published at the end of the workflow;
- provide delete-artifact for a 'cleanup' step at the end of an workflow.
- (EDIT) provide another way to share files/folders between jobs.

**### Describe alternatives you've considered**

Running a scheduled task to clean up artefacts that weren't needed outside their workflows is an unneeded overhead.

**Comment by yacaovsnc** [comment]
Understand the problem.  Currently the closest workaround is to use [retention period](https://github.com/actions/upload-artifact#retention-period) of `1`, so artifacts will be cleaned up after a day.  That would hopefully eliminate the need of customized scheduled clean uptask?

**Comment by konradpabjan** [comment]
Duplicate of https://github.com/actions/upload-artifact/issues/45 so closing out

Basically transient artifacts that are deleted at the end of a run automatically. Such a feature would negate the need for scheduled workflows
